homework/index.js

```diff
@@ -27,18 +12,130 @@
+    const promise = fetchJSON(url);
+    promise
+      .then(response => {
+        buildRepositoryContributors(response, parent);
+      })
+      .catch(error => {
+        createAndAppend('div', parent, { text: error.message, class: 'alert-error' });
+      });
+  }
+
+  function buildRepositoryInfo(RepositoryElement, parent) {
+    const repository = createAndAppend('div', parent, { id: 'repository' });
+    const description = createAndAppend('div', parent, { id: 'description' });
+    const fork = createAndAppend('div', parent, { id: 'fork' });
+    const dateRow = createAndAppend('div', parent, { id: 'date' });
+
+    createAndAppend('label', repository, { text: 'Repository Name: ' });
+    createAndAppend('a', repository, {
+      href: RepositoryElement.html_url,
+      text: RepositoryElement.name,
+      target: '_blank',
+    });
+    createAndAppend('label', description, { text: 'Description: ' });
+    createAndAppend('span', description, { text: RepositoryElement.description });
+    createAndAppend('label', fork, { text: 'Fork: ' });
+    createAndAppend('span', fork, { text: RepositoryElement.forks });
+    createAndAppend('label', dateRow, { text: 'Date: ' });
+    createAndAppend('span', dateRow, { text: RepositoryElement.updated_at });
+  }
+
+  function buildRepositoryList(repositories, parent) {
+    const top = createAndAppend('div', parent, { id: 'top' });
+    const body = createAndAppend('div', parent, { id: 'bodyTag' });
+    const rightSide = createAndAppend('div', body, { id: 'right' });
+    createAndAppend('label', top, { text: 'Select a Repository: ' });
+    const selectRepositoryMenu = createAndAppend('select', top, {
+      id: 'selectMenu',
+      class: 'slate',
+    });
+    repositories.sort((a, b) => a.name.localeCompare(b.name, 'en', { sensitivity: 'base' }));
+    repositories.forEach(repository => {
+      createAndAppend('option', selectRepositoryMenu, {
+        value: repository.id,
+        text: repository.name,
+      });
+    });
+    const repositorySection = createAndAppend('div', body, { id: 'repositoryInfo' });
+    selectRepositoryMenu.addEventListener('change', event => {
+      const selectedRepos = repositories.find(option => {
+        if (Number(option.id) === Number(event.target.value)) {
+          return option;
+        }
+        return undefined;
+      });
+      repositorySection.innerHTML = '';
+      buildRepositoryInfo(selectedRepos, repositorySection);
+      rightSide.innerHTML = '';
+      fetchRepositoryContributors(selectedRepos.contributors_url, rightSide);
+    });
+    // for the first time call.
+    repositorySection.innerHTML = '';
+    buildRepositoryInfo(repositories[0], repositorySection);
+    rightSide.innerHTML = '';
+    fetchRepositoryContributors(repositories[0].contributors_url, rightSide);
+  }
+
+  function main(url) {
+    const promise = fetchJSON(url);
+    promise
+      .then(response => {
+        const root = document.getElementById('root');
```

Comment:
Hi Mohamd - you can also move out the retrieval of `root` element at the beginning of function and need not repeat it in both `.then` and `.catch` blocks.